### ISSUE TITLE
Very slow block updates, not predictable

### ISSUE THREAD

**Issue by AaronMcHale** [SBPrime/AsyncWorldEdit-Premium#113]
Hello

We are using version 3.4.6 but I believe this issue has been happening even before the version we are running.  Sometimes block updates using AsyncWorldEdit happen instantly, sometimes they take a long time, with roughly one block change every second or so.

There doesn't seem to be any predictability to it either, it happens regardless of the number of blocks in the queue, using commands like //paste //set and /brush, even with only a small number of blocks in the queue.  What makes this stranger is it doesn't always happen, as I mentioned sometimes Edits happen instantly, sometimes they are very slow, even Edits of roughly the same size experience this inconsistency.  The Java VM has at least 2 GB of fee memory so it shouldn't be a memory issue and the config is just the standard one that comes with AsyncWorldEdit.

We are using BlocksHub with CoreProtect, but while that could have a performance impact I'd expect it to be consistent and no where near this extreme.  The server doesn't experience any other kind of lag, CoreProtect itself doesn't lag at any6 point and the TPS is sitting at 20.

Any help would be appreciated,
Thank you
Aaron

**Comment by SBPrime** [comment]
1. Is the edit causing lot of light update?
2. What is your AWE config?
3. What is your commandline to start spigot

**Comment by AaronMcHale** [comment]
Regarding Light Updates, light updates don't seem to have an effect just simply replacing a few blocks, generating spheres, and even simply just set on a large area.

The config is pretty standard, haven't changed much of anything, anything I have modified was after noticing the change, but here is my current config: http://pastebin.com/n2gXUeRn

Java arguments used:
-Xmx24576M
-Xms24576M
-server
-XX:+DisableExplicitGC
-XX:+UseParNewGC
-XX:+UseConcMarkSweepGC
-XX:+UseAdaptiveGCBoundary
-XX:-UseGCOverheadLimit
-XX:+AggressiveOpts
-XX:+UseStringCache
-XX:+OptimizeStringConcat
-XX:+UseCompressedOops

**Comment by silverforte** [comment]
I just started having this issue after updating to the most recent version and starting a 1.11 server. It goes fast at first but after a few seconds, grinds to about zero. Takes forever to perform simple tasks that AWE used to handle very efficiently.

**Comment by q220** [comment]
I seem to have a similar issue to this. In the beginning operations happen super fast and after some time it slows down to an almost block-by-block pace. I only noticed when I started doing a very large //replacenear air stone 500 command and at some point I got no more messages. I cancelled the job and then tried a small 50x50 //set stone and then I actually saw it placing block by block.

**Comment by SBPrime** [comment]
I tried to reproduce this without any luck. 
What might help to diagnose the problem: enable debug mode in awe and send me the log. @q220 since you ware able to reproduce this on a small area your case might be the simplest to reproduce. You should be ok to do the first edit with debug off. After that set debug to on and do /awe reload all.

**Comment by q220** [comment]
After having this problem for a few more days, I think this might be related to Dynmap or some deeper lying bug in Spigot. When doing a very large rendering operationg with Dynmap, it seems to grind to a halt where tile updates happen less often. Every time when I noticed this problem there was also the issue with block placement of AWE being really slow. Memory usage was higher than usual, but was not near the maximum. I'm currently doing another really large render with debug mode on to see if it gives any useful information.

**Comment by q220** [comment]
Posting a log where I started an AWE operation after Dynmap was doing an hour of rendering.
I made a small selection of 10500 blocks in total, in this case setting a piece of an ocean to air.

It tried to replace the area block by block. Since the operation would probably take half an hour, I //undo'd after a good minute or so, and also the undo restores block by block.

https://gist.githubusercontent.com/q220/6f4e85b7d05c6f56ca05c1edae428a28/raw/994b478f462a173c4cbb2e73cba00115df2e0454/debug%2520logfile%2520AWE

At some point it nags about Prism, but I verified the prism config file and worldedit logging is off.
